pkg/cache/inmemory.go

```diff
@@ -0,0 +1,99 @@
+package cache
```

Comment:
noob question, why we implement the cache ourself instead of using third party?

Reply:
only because it's a simple in memory cache, do you have any recommended package for in mem cache?

Reply:
can find in [here](https://github.com/avelino/awesome-go#caches), but this one is quite popular
- https://github.com/patrickmn/go-cache
- https://github.com/allegro/bigcache